runtime/lua/vim/lsp/health.lua

```diff
@@ -48,4 +49,33 @@ function M.check()
   end
 end
 
+local function check_watcher()
+  vim.health.start('vim.lsp: File watcher')
+  local watchfunc = vim.lsp._watchfiles._watchfunc
+  assert(watchfunc)
+  local watchfunc_name --- @type string
+  if watchfunc == vim._watch.watch then
+    watchfunc_name = 'libuv-watch'
+  elseif watchfunc == vim._watch.poll then
+    watchfunc_name = 'libuv-poll'
```

Comment:
I think we should rename the method, it's not actually polling anymore but opposed to the watch method it sets up a watcher per directory